bundle-size/api.js

```diff
@@ -206,32 +291,50 @@ exports.installApiRouter = (app, db, userBasedGithub) => {
     };
 
     try {
+      app.log(
+        `Fetching master bundle-sizes on base commit ${baseSha} for pull ` +
+          `request #${check.pull_request_id}`
+      );
       const masterBundleSizes = JSON.parse(
         await getBuildArtifactsFile(github, `${baseSha}.json`)
       );
 
+      app.log(
+        'Fetching mapping of file approvers and thresholds for pull request ' +
+          `#${check.pull_request_id}`
+      );
+      const fileApprovers = await getFileApprovalsDictionary(github, app.log);
+
       // Calculate and collect all (non-zero) bundle size deltas and list all
       // dist files that are missing either from master or from this pull
-      // request.
-      const otherBundleSizeDeltas = [];
+      // request, and all the potential teams that can approve above-threshold
+      // deltas.
+      const bundleSizeDeltas = [];
       const missingBundleSizes = [];
+      const allPotentialApproverTeams = new Set();
       for (const [file, baseBundleSize] of Object.entries(masterBundleSizes)) {
-        if (file === 'dist/v0.js') {
-          // For now, skip dist/v0.js because it gets special handling.
-          continue;
-        }
-
         if (!(file in prBundleSizes)) {
           missingBundleSizes.push(`* \`${file}\`: missing in pull request`);
           continue;
         }
 
         const bundleSizeDelta = prBundleSizes[file] - baseBundleSize;
         if (bundleSizeDelta !== 0) {
-          otherBundleSizeDeltas.push(
+          bundleSizeDeltas.push(
             `* \`${file}\`: ${formatBundleSizeDelta(bundleSizeDelta)}`
           );
         }
+
+        if (
+          file in fileApprovers &&
+          bundleSizeDelta >= fileApprovers[file].threshold
+        ) {
+          // Since `.approvers` is an array, it must be stringified to maintain
+          // the Set uniqueness property.
+          allPotentialApproverTeams.add(
+            JSON.stringify(fileApprovers[file].approvers)
```

Comment:
Nit: just `.toString()` instead

Reply:
I later do `JSON.parse()` on the same entry to convert it back into an array, so `JSON.stringify()` makes more sense here